metrics/encoder.go

```diff
@@ -0,0 +1,67 @@
+package metrics
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Encoder is the signature for encoders
+type Encoder func(name string, op Op, value interface{}, tags Tags, rate float64) (string, error)
+
+// StdoutEncoder is a simple encoder to be used to write to stdout.
+func StdoutEncoder(name string, op Op, value interface{}, tags Tags, rate float64) (string, error) {
+	return fmt.Sprintf("METRIC: %s | %d | %v | %v | %f\n", name, op, value, tags, rate), nil
+}
+
+// StatsDEncoder implements statsd protocol
+func StatsDEncoder(name string, op Op, value interface{}, tags Tags, rate float64) (string, error) {
+	ct := make([]string, len(tags))
+	for k, t := range tags {
+		ct[k] = fmt.Sprintf("%v:%v", t.Key, t.Value)
+	}
+	st := strings.Join(ct, ",")
+
+	switch op {
+	case OpCounterAdd:
+		return fmt.Sprintf("%s:%v|c|@%.4f|#%s\n", name, value, rate, st), nil
+	case OpGaugeUpdate:
+		return fmt.Sprintf("%s:%v|g|@%.4f|#%s\n", name, value, rate, st), nil
+	case OpHistogramUpdate:
+		return fmt.Sprintf("%s:%v|h|@%.4f|#%s\n", name, value, rate, st), nil
+	case OpEventSend:
```

Comment:
Missing almost all the Event args: https://docs.datadoghq.com/api/#events-post

Something like this could work:

```
        properties := []string{}
	encodedTags := []string{}
	for _, t := range tags {
		switch t.Key {
		case "date_happened":
			properties = append(properties, fmt.Sprintf("d:%v", t.Value))
		case "hostname":
			properties = append(properties, fmt.Sprintf("h:%v", t.Value))
		case "aggregation_key":
			aggregationKey = t.Value.(string)
		case "priority":
			properties = append(properties, fmt.Sprintf("p:%v", t.Value))
		case "source_type_name":
			properties = append(properties, fmt.Sprintf("s:%v", t.Value))
		case "event_type":
			properties = append(properties, fmt.Sprintf("t:%v", t.Value))
		default:
			encodedTags = append(encodedTags, fmt.Sprintf("%v:%v", t.Key, t.Value))
		}
	}
```